src/components/Tags/Tags.tsx

```diff
@@ -16,32 +21,28 @@ export interface TagsListProps {
    */
   className?: string;
   /**
-   * Individual Tag class
+   * Child element pass in to display as item content.
    */
-  tagClassName?: string;
+  children?: React.ReactNode;
   /**
-   * List of tags to render
-   */
-  tagsList: string[];
-  /**
-   * `onRemove` callback removes the clicked tag from the list
+   * Individual Tag class
    */
-  onRemove: (tag: string) => void;
+  tagClassName?: string;
```

Comment:
now that you have children, you no longer need this prop as they can just pass it directly to the Tag component child

Reply:
the `onClick` is on a div that's nested in `Tags` how I can tell it which element the `onRemove` function should be attached to?